cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactVersion.java

```diff
@@ -112,7 +112,7 @@ public String getSuffix() {
   }
 
   public boolean isSnapshot() {
-    return suffix != null && "snapshot".equals(suffix.toLowerCase());
+    return suffix != null && suffix.toLowerCase().contains("snapshot");
```

Comment:
actually lets remove this and do it in a separate PR.  It will exacerbate a bug we found today, such that master restarts will break plugins.